pkg/operator/controller/controller.go

```diff
@@ -160,6 +170,16 @@ func (r *ReconcileCDI) Reconcile(request reconcile.Request) (reconcile.Result, e
 
 	// mid delete
 	if cr.DeletionTimestamp != nil {
+		waitForDVs, err := r.waitForDVs(reqLogger, cr)
+		if err != nil {
+			return reconcile.Result{Requeue: true, RequeueAfter: 5 * time.Second}, err
```

Comment:
Any error will automatically requeue the object. So no need to say requeue after 5 seconds.

Reply:
You don't need the Requeue part, because you are returning an error:
```go
return reconcile.Result{}, err
```